### PR TITLE
Fix CI failures on `to_timestamp()` calls

### LINKED ISSUE #7939: CI Failure / Panic when `to_timestamp()` called
### Describe the bug

After merging https://github.com/apache/arrow-datafusion/pull/7844 AND some other PRs I can see the tests and `to_timestamp()` calls are panicking 

```
External error: query failed: DataFusion error: Arrow error: Invalid argument error: column types must match schema types, expected Timestamp(Nanosecond, None) but found Timestamp(Second, None) at column index 0
```

Solely https://github.com/apache/arrow-datafusion/pull/7844 passed all the tests

### To Reproduce

```
select to_timestamp(cast (1 as int));
```

### Expected behavior

No panic

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by comphead**
## Which issue does this PR close?

Closes #7939 .

## What changes are included in this PR?

Fix for `to_timestamp` to support multiple return types depending on the input type

## Are these changes tested?

Yes

### CODE REVIEW

**Comment by comphead** [comment]
@alamb please review

File: datafusion/expr/src/built_in_function.rs [hunk 1]
```diff
@@ -755,7 +755,10 @@ impl BuiltinScalarFunction {
                     return plan_err!("The to_hex function can only accept integers.");
                 }
             }),
-            BuiltinScalarFunction::ToTimestamp => Ok(Timestamp(Nanosecond, None)),
+            BuiltinScalarFunction::ToTimestamp => Ok(match &input_expr_types[0] {
+                dt if dt.is_integer() => Timestamp(Second, None),
```
Comment (alamb) [inline datafusion/expr/src/built_in_function.rs:759]: The implementation seems to  handle `Int64`: https://github.com/apache/arrow-datafusion/blob/ae9a446c41dfa515eb454561c160ad9aa26a7117/datafusion/physical-expr/src/functions.rs#L75-L98

I would expect this code to follow the same semantics. 

So perhaps something like (untested):

```suggestion
                &DataType:Int64 => Timestamp(Second, None),
```


(it looks like UTf8 is already covered by _ as passing a UTF8 results in a timestamp nanosecond)

What would you think about changing the implementation to always return Timestamp(Nanoseconds)? even for an integer argument?

**Review by alamb — COMMENTED** [review]
Thanks @comphead  -- I am not quite sure Int64 vs is_integer -- it might not matter, but it seems strange to me

Also, I wonder if we can simply make `to_timestamp` return `Timestamp::Nanoseconds` always 🤔

File: datafusion/expr/src/built_in_function.rs (hunk 1 above)
Comment (comphead) [reply datafusion/expr/src/built_in_function.rs:759]: Testing integers.
We cannot return always nanoseconds, in that case we limit the output as cannot represent timestamps far in the past or in the future. example https://github.com/apache/arrow-datafusion/issues/7802 and also its incompatible with PG https://github.com/apache/arrow-datafusion/issues/2979

**Review by alamb — APPROVED** [review]
Thank you @comphead for the fast turnaround

**Comment by alamb** [comment]
I took the liberty of pushing a commit to fix a typo (that I introduced 🤦 )

**Pull request merged**